staging/src/k8s.io/legacy-cloud-providers/azure/auth/azure_auth.go

```diff
@@ -132,6 +142,78 @@ func GetServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment) (
 	return nil, ErrorNoAuth
 }
 
+// GetMultiTenantServicePrincipalToken creates a new multi-tenant service principal token based on the configuration
+// PrimaryToken of the result is for the AAD Tenant specified by TenantID, and AuxiliaryToken of the result is for the AAD Tenant specified by CrossTenantNetworkResourceConfig.TenantID
+func GetMultiTenantServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment) (*adal.MultiTenantServicePrincipalToken, error) {
+	if config.CrossTenantNetworkResourceConfig == nil {
+		return nil, fmt.Errorf("crossTenantNetworkResource must be configured when getting multi-tenant service principal token")
+	}
+
+	if strings.EqualFold(config.IdentitySystem, ADFSIdentitySystem) {
+		return nil, fmt.Errorf("getting multi-tenant service principal token doesn't support for ADFS identity system")
+	}
+
+	if config.UseManagedIdentityExtension {
+		return nil, fmt.Errorf("getting multi-tenant service principal token doesn't support for managed identity")
+	}
+
+	multiTenantOAuthConfig, err := adal.NewMultiTenantOAuthConfig(
+		env.ActiveDirectoryEndpoint, config.TenantID, []string{config.CrossTenantNetworkResourceConfig.TenantID}, adal.OAuthOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("creating the multi-tenant OAuth config: %v", err)
+	}
+
+	if len(config.AADClientSecret) > 0 {
+		klog.V(2).Infoln("azure: using client_id+client_secret to retrieve multi-tenant access token")
+		return adal.NewMultiTenantServicePrincipalToken(
+			multiTenantOAuthConfig,
+			config.AADClientID,
+			config.AADClientSecret,
+			env.ServiceManagementEndpoint)
+	}
+
+	if len(config.AADClientCertPath) > 0 && len(config.AADClientCertPassword) > 0 {
+		return nil, fmt.Errorf("getting multi-tenant service principal token doesn't support for AAD Application client certificate authentication")
```

Comment:
`AAD Application client certificate authentication is not supported in getting multi-tenant service principal token`

Reply:
Thank you.